packages/astro/client-base.d.ts

```diff
@@ -0,0 +1,173 @@
+/// <reference types="vite/types/importMeta" />
+
+// CSS modules
+type CSSModuleClasses = { readonly [key: string]: string };
```

Comment:
This was the real magic sauce for ESM import types

Moving out all the types _except_ images into a `client-base.d.ts` allows the main use case of `astro/client` to work.  The image integration now uses only this base file, avoiding the issue of duplicating `*.jpg` module definitions